torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3235,6 +3245,14 @@ def simplify(self, expr: "sympy.Expr") -> "sympy.Expr":
             # divisions simplified away
             if new_pows.issubset(pows) and new_rationals.issubset(rationals):
                 expr = new_expr
+        if expr.has(Mod):
```

Comment:
Let's just change `Mod(x, y)` to:
```
ModularIndexing(x, 1, y)
```
having both ModularIndexing(x, 1, y) and Mod(x, y) in the IR just makes things more complex, since we will need to write the same simplification rule for both.

Reply:
Does it really have to be part of this PR? There's quite a few simplification rules written around `Mod` that would have to be ported to `ModularIndexing` for this to work

Reply:
Looks like guard printing will need a bunch of updates too. I don't think converting all Mods in the IR to ModularIndexing is going to be a trivial change